master/master.c

```diff
@@ -1987,7 +1987,7 @@ static void limit_fds(rlim_t x)
                rl.rlim_cur, rl.rlim_max);
     }
 
-    if (setrlimit(RLIMIT_NUMFDS, &rl) < 0) {
+    if (setrlimit(RLIMIT_NUMFDS, &rl) < 0 && x != RLIM_INFINITY) {
```

Comment:
And this change makes it so that it _won't_ log an error when it tries and fails to make the file descriptors unlimited?  Which seems wrong: if the admin has requested the process should have an unlimited number of fds, and yet the process does not, surely they wish to know about this?